### ISSUE TITLE
Unspent Outputs API. Spent outputs with 0 confirmations are listed in unspent outputs.

### ISSUE THREAD

**Issue by MeLoco1** [coinbase/toshi#185]
I'm using UTXO API to get the outputs to create a transaction.
I created transaction and broadcasted it. Then I created the second tx spending BTC from the same address. 
On attempt to broadcast it I got {"error":"AcceptToMemoryPool() : already spent in the memory pool"}
After first tx had been confirmed I got {"error":"AcceptToMemoryPool() : inputs already spent"}

I reproduce the issue and found that unspent outputs are still listed when they are already spent but not confirmed.

**Comment by aalness** [comment]
That's correct and expected at the moment. The unspent outputs end-point only returns confirmed unspent outputs and similarly won't remove the output until its spend is confirmed. It's definitely something worth fixing.

**Comment by MeLoco1** [comment]
I'm confused. Can you specify.
Is it correct and expected or It's definitely something worth fixing?

I believe it also related to 
https://github.com/coinbase/toshi/issues/177

**Comment by aalness** [comment]
The current behavior is expected but not necessarily incorrect. The behavior could be improved. Yes it's the same issue essentially.

**Comment by MeLoco1** [comment]
Do you have plans to improve the behaviour?
